### PR TITLE
fix: sidebar scroll otel hub

### PR DESCRIPTION
**Pull request by manika-signoz**
## Summary

Refactors the OpenTelemetry Hub so the **sidebar, hub header tabs, progress bar, and outer doc grid** live in a **persistent route-group layout** (`app/(opentelemetry-hub-routes)/layout.tsx` + client shell). Article pages under `/blog`, `/guides`, `/comparisons`, and `/opentelemetry` now render **hub chrome once** and swap **content only**, fixing scroll/layout issues when navigating between hub articles. `layouts/OpenTelemetryHubLayout.tsx` is narrowed to **content-only** (`OpenTelemetryHubContent`). The hub sidebar’s mobile overlay/trigger wiring re-runs when the **normalized route** changes. Hub header tab **active** styling uses the hub **`pathKey`** so the correct tab (e.g. first “Learn” link) shows as selected.

**Stacking:** This branch is intended to stack on **[#2961](https://github.com/SigNoz/signoz.io/pull/2961)** (`chore/ssr-otel-links`).

## Motivation / Problem

On the parent branch, the full hub layout (including sidebar) lived inside the **page-level** hub layout component, so navigating between hub routes could **remount** the shell and break **scroll** / layout expectations. Moving the shell to a **shared layout** for all hub URLs keeps the sidebar and outer structure stable. Mobile sidebar behavior needed to **re-attach** to the DOM when the active hub article changes.

## Changes

Splits OpenTelemetry Hub into a shell (layout + client wrapper) and **per-page content**, moves hub routes into a **route group** that shares that layout, and fixes **sidebar mobile overlay** binding and **header tab** active state for hub navigation.

## Type of Change

- [ ] **Docs** – Changes to `data/docs/**`
- [ ] **Blog** – Changes to `data/blog/**`
- [x] **Site Code** – Changes to `app/**`, `components/**`, `hooks/**`, `utils/**`, config, etc.
- [ ] **Redirects** – Renamed/moved docs or updated `next.config.js` redirects
- [ ] **Dependencies / CI / Scripts**
- [ ] **Other** – Explain below

## Impact

- [ ] Documentation only
- [x] UI changes
- [ ] Developer workflow
- [ ] Performance impact
- [ ] Breaking change

If breaking change, describe migration steps:

N/A - public URLs for hub routes remain `/blog/...`, `/guides/...`, `/comparisons/...`, `/opentelemetry/...`.

## Before / After (if applicable)

https://github.com/user-attachments/assets/b5d08c8e-90ee-42d8-8843-ec28f3374766

https://github.com/user-attachments/assets/24efc375-8edf-4ed7-83f2-8bc1b499679f

### General
- [ ] Branch is up to date with `main`
- [ ] PR title follows conventional format (`type: description`)
- [ ] Self-reviewed the code
- [ ] Comments added for complex logic

### For all changes
- [ ] Built locally (`yarn build`) with no errors
- [ ] Ran `yarn lint` and fixed any issues
- [ ] Pre-commit hooks passed (or ran `yarn check:doc-redirects` / `yarn check:docs-metadata` if applicable)

### For docs changes (`data/docs/**`)
- [ ] Followed the docs author checklist in [contributing/docs-authoring.md](https://github.com/SigNoz/signoz.io/blob/main/contributing/docs-authoring.md)
- [ ] Added/updated the page in `constants/docsSideNav.ts` if adding or moving a doc

### For blog changes
- [ ] Followed the blog workflow in [contributing/blog-workflow.md](https://github.com/SigNoz/signoz.io/blob/main/contributing/blog-workflow.md)
- [ ] Frontmatter includes `title`, `date`, `author`, `tags` (and `canonicalUrl` if applicable)
- [ ] Images use WebP format and live under `public/img/blog/<YYYY-MM>/`

### For site code changes
- [ ] Followed the site code playbook in [contributing/site-code.md](https://github.com/SigNoz/signoz.io/blob/main/contributing/site-code.md)
- [ ] New dependencies are justified in the PR description (if any)

### For renamed or moved docs
- [ ] Followed the redirects and discovery section in [contributing/docs-authoring.md](https://github.com/SigNoz/signoz.io/blob/main/contributing/docs-authoring.md)
- [ ] Added permanent redirect in `next.config.js` under `async redirects()`
- [ ] Updated internal links and sidebar in `constants/docsSideNav.ts`
- [ ] Ran `yarn check:doc-redirects` to verify

## Testing

- [ ] Tested locally
- [ ] Edge cases considered
- [ ] Existing functionality verified
- [ ] New tests added (if applicable)

Steps to test:
Open a hub article under **Learn OpenTelemetry** (e.g. a blog post in the hub), then navigate to another hub article via the **sidebar** — confirm the **sidebar stays mounted**, scroll feels correct, and the **active tab** matches the current hub section.

## Rollout / Deployment Notes
ensure deployment order respects stacking: [#2961](https://github.com/SigNoz/signoz.io/pull/2961) first.

## Additional Context

Stacked from [#2961](https://github.com/SigNoz/signoz.io/pull/2961)

---

**Note:** Submit as Draft by default. Mark "Ready for review" when checks pass and content is ready.

### CODE REVIEW

File: layouts/OpenTelemetryHubLayout.tsx [hunk 1]
```diff
@@ -96,30 +92,20 @@ const getFormattedDate = (content: LayoutProps['content']) => {
     : null
 }
 
-export default function OpenTelemetryHubLayout({
+/**
+ * Content-only hub component. The outer shell (sidebar, header tabs, grid
+ * container) lives in `components/OpenTelemetryHubShell.tsx` (mounted in the
+ * root layout) so the sidebar persists across all route-segment navigations.
+ */
```
[the remark this answers is not in the document]
Comment (manika-signoz) [reply layouts/OpenTelemetryHubLayout.tsx:99]: done

File: utils/opentelemetryHub.ts [hunk 2]
```diff
@@ -221,8 +221,12 @@ function getHubIndex(): HubIndex {
   return buildHubIndex()
 }
 
+const ROUTE_ALIASES: Record<string, string> = {
+  '/opentelemetry': '/blog/what-is-opentelemetry',
+}
+
 export function getHubContextForRoute(route: string) {
-  const normalized = normalizeRoute(route)
+  const normalized = ROUTE_ALIASES[normalizeRoute(route)] ?? normalizeRoute(route)
```
[the remark this answers is not in the document]
Comment (manika-signoz) [reply utils/opentelemetryHub.ts]: done

File: layouts/open-telemetry-hub/OpenTelemetrySidebarClient.tsx [hunk 3]
```diff
@@ -58,7 +58,7 @@ export default function OpenTelemetrySidebarClient({
   useEffect(() => {
     setTriggerContainer(document.getElementById(mobileTriggerId))
     setOverlayContainer(document.getElementById(mobileOverlayId))
-  }, [mobileOverlayId, mobileTriggerId])
+  }, [mobileOverlayId, mobileTriggerId, normalizedRoute])
```
[the remark this answers is not in the document]
Comment (manika-signoz) [reply layouts/open-telemetry-hub/OpenTelemetrySidebarClient.tsx:61]: fine for now

File: app/(opentelemetry-hub-routes)/layout.tsx [hunk 4]
```diff
@@ -0,0 +1,139 @@
+'use client'
+
```
[the remark this answers is not in the document]
Comment (manika-signoz) [reply app/(opentelemetry-hub-routes)/layout.tsx]: removed use client from layout.tsx and moved to a client wrapper

**Review by YounixM — DISMISSED** [review]

**Review by YounixM — APPROVED** [review]
